src/JITModule.cpp

```diff
@@ -512,31 +512,93 @@ void JITModule::make_externs(const std::vector<JITModule> &deps, llvm::Module *m
 }
 
 void *JITModule::main_function() const {
-    if (!defined()) {
```

Comment:
It is no longer possible to construct a JITModule without initializing JITModuleContents and there is no way to reset the IntrusivePtr so it seems a worthless assert.